### ISSUE TITLE
[FUNCTIONALITY]: Explore the Table component for HTML structure improvements 

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/vets-design-system-documentation#489]
# Bug Report
- [ ] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened
The [Table component](https://github.com/department-of-veterans-affairs/component-library/blob/master/src/components/Table/Table.jsx) *could* benefit from using a `th[scope="row"]` for the first element in each row. Screen readers get additional context when we scope rows appropriately, like they do when we scope columns in the table heading.

## Acceptance Criteria
- [ ] Team reviews the possible fix and opts to fix or close the ticket as "won't fix"
- [ ] 0 axe violations if HTML is updated
- [ ] Create and/or modify unit tests for this component to include an aXe scan
- [ ] Visual styles remain consistent and functionality is not broken on VA.gov (all tests continue to pass)

## Possible Fixes (optional)
This example uses proper `scope` attributes and a `<th>` for first table row element. This is the preferred means to create table structure, and is well-supported by assistive technology.

```diff
<table>
  <thead>
    <tr>
      <th scope="col">Name</th>
      <th scope="col">Email</th>
      <th scope="col">Message</th>
    </tr>
  </thead>
  <tbody>
    <tr>
+     <th scope="row">Joe</th>
      <td>joe@joe.com</td>
      <td>Hello</td>
    </tr>
    <tr>
+     <th scope="row">Bob</th>
      <td>bob@bob.com</td>
      <td>Hola</td>
    </tr>
    <tr>
+     <th scope="row">Tom</th>
      <td>tom@tom.com</td>
      <td>What's Up</td>
    </tr>
  </tbody>
</table>
```

## Screenshots or Trace Logs
![Screen Shot 2020-05-18 at 10 39 35 AM](https://user-images.githubusercontent.com/934879/82232397-32cd9f00-98f4-11ea-8ab3-91c24eaa434e.png)
